ct: rename min_epoch to start_epoch in CT GC

dotnwat opened the pull request:
The original name was rather confusing because it was computed as max, but the name made it seem like it was min.

<!--
See https://github.com/redpanda-data/redpanda/blob/dev/CONTRIBUTING.md#pull-request-body
for more details and examples of what is expected in a PR body.

Content in this top section is REQUIRED. Describe, in plain language, the motivation
behind the change (bug fix, feature, improvement) in this PR and how the included
commits address it.

Add the GitHub keyword `Fixes` to link to bug(s) this PR will fix, e.g.
  Fixes #ISSUE-NUMBER, Fixes #ISSUE-NUMBER, ...

If this PR is a backport, link to the original with `Backport of PR`, e.g.
  Backport of PR #PR-NUMBER
-->

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [x] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v25.2.x
- [ ] v25.1.x
- [ ] v24.3.x

## Release Notes

<!--
If the changes in this PR do not need to be mentioned in the release
notes, then don't add a sub-section and simply list `none`, e.g.

Otherwise, adding a sub-section or `none` is REQUIRED if the PR is not a backport PR.
If this is a backport PR, adding contents to this section will override
the release notes section inherited from the original PR to dev.

Add one or more of the sub-sections with a short description bullet
point of the change, e.g.

### Bug Fixes

* Short description of the bug fix if this is a PR to `dev` branch.

### Features

* Short description of the feature. Explain how to configure.

### Improvements

* Short description of how this PR improves existing behavior.

-->

* none

andrwng commented on src/v/cloud_topics/level_zero/batcher/aggregator.cc:
```diff
@@ -149,7 +149,7 @@ void aggregator<Clock>::add(l0::write_request<Clock>& req) {
     it->second.push_back(req);
     _size_bytes += req.size_bytes();
-    _min_epoch = std::max(_min_epoch, req.min_epoch);
+    _highest_min_epoch = std::max(_highest_min_epoch, req.min_epoch);
```
Thanks, thoughts on the followup comment?

https://github.com/redpanda-data/redpanda/pull/28181#discussion_r2461686661

where instead of `req.min_epoch` it's `req.topic_start_epoch` and here it's `highest_topic_start_epoch`?

This only has properties of a "minimum" value in the context of the batcher call where it's used as a lower bound, so IMO in all these other places, it feels more intuitive to just name this what it is, not how it's intended on being used.

dotnwat replied:
I think the issue with using "topic_start" in the name is that it's an implementation detail that we use the topic revision id at all, and we may very well move to a different source for this epoch.

andrwng replied:
It feels kind of okay to me to expose that implementation detail (especially if it makes the code less surprising), given the write request, aggregator, and batcher are kind of tightly coupled, vs this being a more widely used public interface. Is there a different source of the epoch that you have in mind?

andrwng replied:
Or maybe just `start_epoch`?

dotnwat replied:
how's that in the latest push?

andrwng replied:
Awesome, thanks for the change

dotnwat committed:
ct: use topic start epoch naming in L0 GC

Instead of trying to completely hide the source of a CTP min epoch
value (which has lead to confusing names) we can just lean in to it a
bit since it's internal anyway and it makes things much simpler to
understand.

Signed-off-by: Noah Watkins <noahwatkins@gmail.com>

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/74972-->
<!--589cabb7-dd8d-4f80-8e2c-1f1262b8fd76-->
<!--title-->
## Retry command for [Build#74972](https://buildkite.com/redpanda/redpanda/builds/74972)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/tls_version_test.py::TLSVersionTestRSA.test_change_version@{"version":3}
```
<!--/body-->

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--64b1995e-781e-43c6-915c-dcc2dbfb4e45-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/74972">build#74972</a></summary>


| test_class                          | test_method                     | test_arguments              | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                   | test_history                                                                                                                                                               |
|:------------------------------------|:--------------------------------|:----------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:---------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| ShadowLinkBasicTests                | test_rapid_shadow_link_toggling | null                        | integration | https://buildkite.com/redpanda/redpanda/builds/74972#019a17f1-ceef-4543-99cf-a0a9d4862975 | FLAKY         | 14/21    | upstream reliability is '91.16883116883116'. current run reliability is '66.66666666666666'. drift is 24.50216 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkBasicTests&test_method=test_rapid_shadow_link_toggling         |
| DataTransformsLoggingMetricsTest    | test_logger_metrics_present     | null                        | integration | https://buildkite.com/redpanda/redpanda/builds/74972#019a17f1-cef2-41cd-86ee-aee03bec233b | FLAKY         | 20/21    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS                | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DataTransformsLoggingMetricsTest&test_method=test_logger_metrics_present |
| TLSMetricsTest                      | test_labels                     | null                        | integration | https://buildkite.com/redpanda/redpanda/builds/74972#019a17f3-d73d-4cee-8d0a-b297579c7240 | FLAKY         | 9/21     | upstream reliability is '68.31210191082803'. current run reliability is '42.857142857142854'. drift is 25.45496 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=TLSMetricsTest&test_method=test_labels                                   |
| TLSVersionTestRSA                   | test_change_version             | {"version": 3}              | integration | https://buildkite.com/redpanda/redpanda/builds/74972#019a17f1-cef5-4566-ab44-1e2e26b5b89e | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                   | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=TLSVersionTestRSA&test_method=test_change_version                        |
| WriteCachingFailureInjectionE2ETest | test_crash_all                  | {"use_transactions": false} | integration | https://buildkite.com/redpanda/redpanda/builds/74972#019a17f3-d73c-478e-aa28-483154a98a68 | FLAKY         | 18/21    | upstream reliability is '82.30353929214156'. current run reliability is '85.71428571428571'. drift is -3.41075 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=WriteCachingFailureInjectionE2ETest&test_method=test_crash_all           |
</details>
<!--/body-->
